rabbitmq/tests/test_unit.py

```diff
@@ -103,7 +103,7 @@ def test__get_metrics_3_1(check, aggregator):
 
 
 @pytest.mark.unit
-@mock.patch.object(datadog_checks.rabbitmq.RabbitMQ, '_get_object_data')
+@mock.patch.object(datadog_checks.rabbitmq.rabbitmq.RabbitMQManagement, '_get_object_data')
```

Comment:
@alopezz @jose-manuel-almaza this is an example of why I don't like `mock.patch`: the tests know too much about the implementation when they can grab deep into it and select classes/functions to mock.

Reply:
We could remove the patch decorator and patch inside the function, and either get the class dynamically or patch the object itself. That should make the test more resistant to change.

I personally tend to use [`monkeypatch`](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html) for this, but you could do the equivalent with the other alternatives (like the `mock.patch` decorator), but the gist of it would be to do something like this in the body of the test:

```python

# Patch it on the class, finding the class dynamically:
monkeypatch.setattr(check.__class__, '_get_object_data', mock.Mock(return_value = data))
# or patch it on the object directly (there may be caveats regarding method binding for this in general, but they don't apply for this simple case)
monkeypatch.setattr(check, '_get_object_data', mock.Mock(return_value = data))
# or you should be able to even get away with (since `check` is supposed to be a fresh instance that won't be used elsewhere)
check._get_object_data = mock.Mock(return_value = data)
```

Reply:
I have made a note of this and will tackle it in a follow-up PR, thanks!